### PR TITLE
Add Ingress support

### PR DESCRIPTION
**Pull request by inverse**
# Proposed Changes

First time trying to do this so any advice would be helpful.

However getting 404 when trying to access the main page. Any idea how to troubleshoot this?

## Related Issues

> ([Github link][autolink-references] to related issues or pull requests)

[autolink-references]: https://help.github.com/articles/autolinked-references-and-urls/

### CODE REVIEW

File: uptime-kuma/Dockerfile [hunk 1]
```diff
@@ -25,10 +25,10 @@ RUN \
         python3=3.11.6-r0 \
         iputils=20221126-r2 \
         setpriv=2.38.1-r8 \
+        git \
     \
     && mkdir -p /opt/uptime-kuma \
-    && curl -L -s "https://github.com/louislam/uptime-kuma/archive/refs/tags/${UPTIME_KUMA_VERSION}.tar.gz" \
-        | tar zxvf - --strip-components 1 -C /opt/uptime-kuma \
+    && git clone --depth 1  --branch add-ips-to-allow  https://github.com/inverse/uptime-kuma.git /opt/uptime-kuma \
```
Comment (inverse) [inline uptime-kuma/Dockerfile:31]: Right now it's a PR on their repo: https://github.com/louislam/uptime-kuma/pull/3981

But I wanted to try and get this working as I saw it was a requirement to restrict access to `172.30.32.2`.

File: uptime-kuma/rootfs/etc/s6-overlay/s6-rc.d/uptime-kuma/run [hunk 2]
```diff
@@ -10,4 +10,7 @@ cd /opt/uptime-kuma || bashio::exit.nok "Could not change directory to Uptime Ku
 
 bashio::log.info "Starting Uptime Kuma..."
 
-exec node server/server.js --data-dir="/data/"
+export UPTIME_KUMA_DB_TYPE=sqlite
+export UPTIME_KUMA_DB_NAME=kuma.db
```
Comment (inverse) [inline uptime-kuma/rootfs/etc/s6-overlay/s6-rc.d/uptime-kuma/run:14]: These two are required as they changed the way the DB is setup on master https://github.com/louislam/uptime-kuma/pull/2738/files#diff-c3ca73872fa7aee135c0942cff2eab343c8137fdb5b1d72a7d41ef9effbf9f16R49

File: uptime-kuma/rootfs/etc/s6-overlay/s6-rc.d/uptime-kuma/run [hunk 3]
```diff
@@ -10,4 +10,7 @@ cd /opt/uptime-kuma || bashio::exit.nok "Could not change directory to Uptime Ku
...
+
+exec node server/server.js --data-dir="/data/" --ips-to-allow="::ffff:172.30.32.2"
```
Comment (inverse) [inline uptime-kuma/rootfs/etc/s6-overlay/s6-rc.d/uptime-kuma/run:16]: For me only worked with this hybrid IP notation

**Comment by frenck** [comment]
Closing this, as this is not possible with the current stable release of Uptime Kuma.

../Frenck

**Pull request closed without merging**